suite.xml SyncRequestContributor

kaapstorm opened the pull request:
WIP. Parallels @proteusvacuum's work in uatbc/case-search-es and uatbc/case-fixture. 

This branch builds the request-sync node for suite.xml. It includes stubs for the OTA views to set the URLs, and Module search_config attribute to determine what search keys the device must pass.

@proteusvacuum @czue cc @gcapalbo

biyeun committed:
migrate care pathways reports to b3

kaapstorm committed:
Add toggle

kaapstorm committed:
view stubs

kaapstorm committed:
suite.xml models

kaapstorm committed:
case search models

kaapstorm committed:
case search models

kaapstorm committed:
SyncRequestContributor

dimagimon commented on corehq/apps/ota/views.py:
```diff
@@ -32,6 +34,52 @@ def restore(request, domain, app_id=None):
+    search_es = CaseSearchES()
+    search_es = search_es.domain(domain)
+    search_es = search_es.case_type(case_type)
+    fuzzies = get_fuzzy_properties(domain, case_type)  # TODO: define
```
F821 undefined name 'get_fuzzy_properties'

dimagimon commented on corehq/apps/ota/views.py:
```diff
@@ -32,6 +34,52 @@ def restore(request, domain, app_id=None):
+@json_error
+@login_or_digest_or_basic_or_apikey()
+def claim(request, domain):
+    couch_user = CouchUser.from_django_user(request.user)
```
F841 local variable 'couch_user' is assigned to but never used

snopoke replied:
I think you can use `request.couch_user`

dimagimon commented on corehq/apps/app_manager/tests/test_suite_sync_request.py:
```diff
@@ -0,0 +1,28 @@
+from django.test import SimpleTestCase
+
+from corehq.apps.app_manager.const import APP_V2
+from corehq.apps.app_manager.models import Application, Module, CaseTypeFuzzyProperties
```
F401 'CaseTypeFuzzyProperties' imported but unused

dimagimon commented on corehq/apps/ota/views.py:
```diff
@@ -32,6 +34,52 @@ def restore(request, domain, app_id=None):
+    for key, value in criteria.items():
+        search_es = search_es.case_property_query(key, value, fuzzy=(key in fuzzies))
+    results = search_es.values()
+    fixtures = CaseDBFixture(  # TODO: Wait, and rebase.
```
F821 undefined name 'CaseDBFixture'

kaapstorm commented:
@snopoke, let's pick this up on Monday.

kaapstorm committed:
First simple test

dimagimon commented on corehq/apps/ota/views.py:
```diff
@@ -32,6 +35,59 @@ def restore(request, domain, app_id=None):
+    for key, value in criteria.items():
+        search_es = search_es.case_property_query(key, value, fuzzy=(key in fuzzies))
+    results = search_es.values()
+    fixtures = CaseDBFixture([CommCareCase.wrap(case) for case in results])  # TODO: Wait, and rebase.
```
F821 undefined name 'CaseDBFixture'

proteusvacuum replied:
Looks good, however, this won't work with SQL cases. (check discussion here: https://github.com/dimagi/commcare-hq/pull/10981#discussion_r58247742).

kaapstorm commented on corehq/apps/app_manager/tests/data/suite/sync_request.xml:
```diff
@@ -0,0 +1,19 @@
+  <sync-request>
+    <post url="https://www.example.com/a/test_domain/phone/claim-case/">
+      <data key="case_id" ref="session/datum/case[@id]"/>
+      <data key="case_type" ref="session/datum/case/type"/>
```
@phillipm @ctsims I'm making a bunch of assumptions re the sync-request node that the device expects in suite.xml. Can you give me some examples of what `ref` needs to look like? From what I understand this need to point to case ID and case type of the search result that the user has chosen.

kaapstorm commented on corehq/apps/app_manager/tests/data/suite/sync_request.xml:
```diff
@@ -0,0 +1,19 @@
+    <session>
+      <query url="https://www.example.com/a/test_domain/phone/search/">
+        <prompt key="name">
+          <display>Name</display>
```
I'm sure this is supposed to look like ...

    <display>
        <text>
            <locale id="something.somethingelse"/>
        </text>
    </display>

... but I'm not sure what the somethings should be in this case. Where do I store what the app builder wants the label to be (and translations), and how do I refer to it here?

snopoke replied:
This should be a proper `display` block: https://github.com/dimagi/commcare/wiki/Suite20#display

snopoke replied:
See `TextOrDisplay` class in python.

snopoke replied:
the somethings are things that app builder makes up and reference values that get put into the app strings files. See `app_strings.py`

snopoke commented on corehq/apps/app_manager/tests/data/suite/sync_request.xml:
```diff
@@ -0,0 +1,19 @@
+        </prompt>
+        <data key="case_type" ref="'case'"/>
+      </query>
+    </session>
```
missing `datum` block

snopoke commented on corehq/apps/app_manager/suite_xml/sections/sync_requests.py:
```diff
@@ -0,0 +1,86 @@
+                <datum/>
+            </session>
+            <stack/>
+        </sync-request>
```
Rather than add this in here it should get added to the specs on github and here you can just use a reference: https://github.com/dimagi/commcare/wiki/Suite20

snopoke replied:
Rather than add this in here it should get added to the specs on github and here you can just use a reference: https://github.com/dimagi/commcare/wiki/Suite20

kaapstorm committed:
Use id_strings for localization

dimagimon commented on corehq/apps/app_manager/suite_xml/sections/sync_requests.py:
```diff
@@ -0,0 +1,86 @@
+from corehq.apps.app_manager import id_strings
+from corehq.apps.app_manager.suite_xml.contributors import SuiteContributorByModule
+from corehq.apps.app_manager.suite_xml.xml_models import (
```
F401 'SessionDatum' imported but unused
F401 'Stack' imported but unused

kaapstorm committed:
Refer docstrings to sync-request defn

kaapstorm commented:
Added updates. A few details on `sync-request` waiting on @phillipm or @ctsims -- especially whether/how `<command>` should refer to the module or case list that Search needs to appear in, and how to refer to chosen search result in `<post>`. See [sync_request.xml][1] for my appalling guesses.


  [1]: https://github.com/dimagi/commcare-hq/blob/suitexml_search/corehq/apps/app_manager/tests/data/suite/sync_request.xml#L7

proteusvacuum commented on corehq/apps/ota/views.py:
```diff
@@ -32,6 +35,54 @@ def restore(request, domain, app_id=None):
+                referenced_type=claim_case_type,
+                relationship='extension'
+            )])
+        claim_delegate.save()
```
Does this work like this? wondering if you have to do `post_case_block

snopoke replied:
Definitely need to use `post_case_block`

proteusvacuum commented on corehq/apps/ota/views.py:
```diff
@@ -32,6 +35,54 @@ def restore(request, domain, app_id=None):
+        claim_delegate = CommCareCase(
+            domain=domain,
+            owner_id=claimant_id,
+            case_type='claim',  # TODO: Really?
```
seems fine for now. Might want to make it a constant?

phillipm commented:
@kaapstorm, I'm just getting spun back up on this, but regarding
> Added updates. A few details on sync-request waiting on @phillipm or @ctsims -- especially whether/how <command> should refer to the module or case list that Search needs to appear in, and how to refer to chosen search result in <post>. See sync_request.xml for my appalling guesses.

The `<post>` should refer to instances using the normal `instance('session')` or `instance('casedb')` syntax.

The `<command>` format hasn't really been nailed down, but will be pretty sparse, it just needs a uniform id, like `<command id="case-claim"> ... <text><local>...`. All the work of referencing case lists is collected in the `<session>` block, which enumerates all the data we need to collect before performing the post action.

phillipm commented:
Forgot to mention, an example of the `sync-request` block can be found in the description of this PR: https://github.com/dimagi/commcare/pull/250

kaapstorm commented:
Thanks @phillipm, that example is excellent!

If I have two modules, and one searches for mothers based on name, and the other searches for mothers based on, say, library card number, I'm still hazy on how to specify which sync-request to use for which module.

phillipm commented:
@kaapstorm you're right, we can't just do `<command id="case-claim">`. Let me get back to you on that

proteusvacuum commented on corehq/apps/app_manager/tests/data/suite/sync_request.xml:
```diff
@@ -0,0 +1,34 @@
+      </display>
+    </command>
+    <session>
+      <query url="https://www.example.com/a/test_domain/phone/search/">
```
@kaapstorm, I think we need to add a `data` node here that fetches the case type from the session. 
From what I've understood from what @phillipm told me, the `data` nodes are sent back as get request parameters. 
i.e. something like:
```
<data key="case_type" ref="instance('session')/session/data/case_type"/>
```
Need to verify how we can get the case_type from the session, or if we just hard-code it ourselves in HQ from the information we know about the module.

kaapstorm committed:
case claim view uses post_case_blocks

kaapstorm committed:
CaseSearch model must support multiple languages

kaapstorm committed:
claim case test

kaapstorm committed:
case search models: constants for suite.xml

kaapstorm committed:
update test

kaapstorm committed:
Update SyncRequestContributor based on feedback

dimagimon commented on corehq/apps/ota/views.py:
```diff
@@ -32,6 +40,65 @@ def restore(request, domain, app_id=None):
+    for key, value in criteria.items():
+        search_es = search_es.case_property_query(key, value, fuzzy=(key in fuzzies))
+    results = search_es.values()
+    fixtures = CaseDBFixture([CommCareCase.wrap(case) for case in results])  # TODO: Wait, and rebase.
```
F821 undefined name 'CaseDBFixture'

kaapstorm committed:
search view uses CaseSearchConfig and CaseDBFixture

kaapstorm committed:
search view uses CaseSearchConfig and CaseDBFixture

kaapstorm commented on corehq/apps/ota/views.py:
```diff
@@ -32,6 +42,70 @@ def restore(request, domain, app_id=None):
+        fixtures = CaseDBFixture([CommCareCase.wrap(case) for case in results])
+    else:
+        cases = (CommCareCaseSQLSerializer(data=case).validated_data for case in results)
+        fixtures = CaseDBFixture([CommCareCaseSQL(**case) for case in cases])
```
@proteusvacuum @snopoke could you just confirm that this is a reasonable way to deserialise ES results?

snopoke replied:
Haven't actually tried this before but a few observations:
* the if else body's should be reversed
* `CommCareCaseSQLSerializer(data=case)` should return a SQL case so no need to go `CommCareCaseSQL(**case)`
* too many different variables called `case`

kaapstorm replied:
1. Lol. Thank you.
2. :+1: That sounds a lot more reasonable! (Django models have a `validated_data` property? I've never used that.)
3. Fair enough.

kaapstorm replied:
Actually, it seems [from the docs](http://www.django-rest-framework.org/api-guide/serializers/#accessing-the-initial-data-and-instance) that instantiating a serializer returns a serializer. You need to populate its instance (which `save()`, `update()` and `create()` do with

    for attr, value in validated_data.items():
        setattr(instance, attr, value)

), and then the populated instance will be available as `serializer.instance`

kaapstorm committed:
get a deserialized instance from serialized data

kaapstorm committed:
Fixed CaseDBFixture instantiation

kaapstorm commented:
Review buddy @millerdev, this is what I've been up to. :briefcase: :mag: Originally opened for review when Giovanni was review buddy.

kaapstorm commented:
(I'm planning to break this up a bit. There are at least two small changes that don't really belong in here, and I'll PR them separately, and merge/rebase them in after they're merged.)

snopoke commented on corehq/apps/app_manager/models.py:
```diff
@@ -2117,7 +2125,7 @@ class Module(ModuleBase, ModuleDetailsMixin):
     task_list = SchemaProperty(CaseList)
     parent_select = SchemaProperty(ParentSelect)
-
+    search_config = SchemaListProperty(CaseSearch)
```
Instead of being a list of properties this should probably be a nested object that contains the list so that you can also have a field for things like title

snopoke commented on corehq/apps/app_manager/suite_xml/sections/sync_requests.py:
```diff
@@ -0,0 +1,115 @@
+        from corehq.apps.app_manager.models import AdvancedModule, Module
+
+        if isinstance(module, (Module, AdvancedModule)) and module.search_config:
+            domain = module.get_app().domain
```
`module.get_app()` -> `self.app`

snopoke commented on corehq/apps/app_manager/suite_xml/sections/sync_requests.py:
```diff
@@ -0,0 +1,115 @@
+                    data=[
+                        QueryData(
+                            key='case_id',
+                            ref="instance('results')/case[@case_type='{}'][@status='open'][0]/@case_id".format(
```
Could be better to make `CaseSelectionXPath` able to use a different instance name

snopoke commented on corehq/apps/app_manager/suite_xml/sections/sync_requests.py:
```diff
@@ -0,0 +1,115 @@
+                    queries=[
+                        SyncRequestQuery(
+                            url=absolute_reverse('sync_search', args=[domain]),
+                            storage_instance='results',
```
requests could be a constant

snopoke commented on corehq/apps/ota/tests/claim_case.py:
```diff
@@ -0,0 +1,49 @@
+            owner_id='someone_else',
+        ).as_xml()], {'domain': DOMAIN})
+
+    def test_claim_case(self):
```
`@run_with_all_backends`

snopoke commented on corehq/apps/ota/tests/claim_case.py:
```diff
@@ -0,0 +1,49 @@
+        """
+        A claim case request should create an extension case
+        """
+        self.assertEqual(len(get_case_ids_in_domain(DOMAIN, type=CLAIM_CASE_TYPE)), 0)
```
`CaseAccessors(domain).get_case_ids_in_domain...)`

snopoke commented on corehq/apps/ota/views.py:
```diff
@@ -32,6 +42,69 @@ def restore(request, domain, app_id=None):
+            if should_use_sql_backend(domain):
+                case = CommCareCaseSQL.objects.get(case_id=host_id)
+            else:
+                case = CommCareCase.get(host_id)
```
Can replace the if else with this `case = CaseAccessors(domain).get_case(host_id)`

snopoke commented on corehq/form_processor/serializers.py:
```diff
@@ -5,6 +5,21 @@
+    ModelClass = SerializerClass.Meta.model
+    serializer = SerializerClass(data=data)
+    assert serializer.is_valid(), 'Unable to deserialize data while creating {}'.format(ModelClass)
+    return ModelClass(**serializer.validated_data)
```
A unit test for this would be great

kaapstorm committed:
CaseTypeXpath.case takes instance name

kaapstorm committed:
Case search config lets you set the command label

kaapstorm committed:
Use CaseTypeXpath with instance name

kaapstorm committed:
Tests support both DB backends

proteusvacuum committed:
return fixture as a string

dimagimon commented on corehq/apps/ota/views.py:
```diff
@@ -32,6 +42,69 @@ def restore(request, domain, app_id=None):
+        search_es = search_es.case_property_query(key, value, fuzzy=(key in fuzzies))
+    results = search_es.values()
+    if should_use_sql_backend(domain):
+        fixtures = CaseDBFixture([get_instance_from_data(CommCareCaseSQLSerializer, result) for result in results]).fixture
```
E501 line too long (123 > 115 characters)

proteusvacuum committed:
Flatten search results to be used in serializers

kaapstorm committed:
Case search endpoint test added

dimagimon commented on corehq/apps/ota/tests/test_search_claim_endpoints.py:
```diff
@@ -0,0 +1,60 @@
+from corehq.apps.domain.shortcuts import create_domain
+from corehq.apps.users.models import CommCareUser
+from corehq.form_processor.interfaces.dbaccessors import CaseAccessors
+from corehq.form_processor.tests import run_with_all_backends
```
F401 'run_with_all_backends' imported but unused

wpride commented:
FYI this is breaking building http://manage.dimagi.com/default.asp?223512#1126039

proteusvacuum committed:
cleanup

proteusvacuum committed:
Copy old extension index tree to new sync long

biyeun committed:
fix size of globe

biyeun committed:
change globe to custom globe

orangejenny committed:
Display double management form options in correct language

orangejenny committed:
Update reports.py

Commit:
get_json -> to_api_json

Commit:
remove attachment_key terminology

orangejenny committed:
sort by date

orangejenny committed:
Update staging.yaml

esoergel committed:
Fix save button's subscription to filter state

I'm not entirely sure how this used to work - it broke with the knockout
3 upgrade.  The docs said a lot changed around bindings - my guess is
2.x would evaluate the binding values even if the name wasn't
recognized, and 3.x ignores the values.  At any rate, I believe this
accomplishes roughly the same thing (And it makes the save button
enable on filter change again).

[ci skip]

Commit:
convert parent_ids to list

orangejenny committed:
Merge pull request #11258 from dimagi/subscribe-filters

Fix save button's subscription to filter state

orangejenny committed:
Merge pull request #11255 from dimagi/fix-globe

Make Globe more Global

orangejenny committed:
Update staging.yaml

orangejenny committed:
Update staging.yaml

Commit:
useLabels and useIds for export display

Commit:
Fix gunicorn command

Commit:
Fix column definition

czue committed:
Merge pull request #11256 from dimagi/extension-sync-bug

Extension sync bug

snopoke committed:
Merge pull request #11253 from dimagi/orangejenny-patch-3

Don't artificially limit supply points

czue committed:
Merge pull request #11254 from dimagi/app-builder-stuff

purge_report_from_mobile_ucr on report deletion

czue committed:
don't use datatables on old reports either

czue committed:
imports

czue committed:
make assert soft

snopoke committed:
sharding setup docs

Commit:
Fixed charts in base b3 template

Commit:
Migrated care reports to b3

czue committed:
Merge pull request #11261 from dimagi/sharding_doc

sharding setup docs

snopoke committed:
move retry handler to parent method

snopoke committed:
catch django and psycopg2 errors

snopoke committed:
Merge pull request #11260 from dimagi/soft-assert

make assert soft

snopoke committed:
Merge pull request #11259 from dimagi/mc-report-fix

don't use datatables on old reports either

snopoke committed:
update readme to use fab manage

snopoke committed:
update fab manage to use task arg

snopoke committed:
return configs

czue committed:
Merge pull request #11263 from dimagi/retry-on-fail

fix retry on connection failure

czue committed:
include current month in dashboard

orangejenny committed:
B3 updates for scheduled_report_form.js

orangejenny committed:
Allow sorting of SMS parts

Commit:
better CommCareIndex rep

Commit:
use get_cases instead of iter_cases

Commit:
unimplement parent function

Commit:
[ci skip] dimagimon

proteusvacuum commented on corehq/apps/app_manager/suite_xml/sections/sync_requests.py:
```diff
@@ -0,0 +1,129 @@
+                                 .select(u'@status', u'open', quote=True)
+                                 .select_raw(0)
+                                 .slash(u'@case_id')),
+                            # e.g. instance('results')/results/case[@case_type='foo'][@status='open'][0]/@case_id
```
I'm not sure this is right. The `<data>` blocks just add default parameters to the GET request to the URL. We would only want `case_type` here, as this is always required when searching.

proteusvacuum replied:
Actually, ignore me.

proteusvacuum commented on corehq/apps/case_search/models.py:
```diff
@@ -3,6 +3,12 @@
+CLAIM_CASE_TYPE = 'claim'
+CALCULATED_DATA = 'calculated_data'  # Datum ID to mark a case as claimed
+MARK_AS_CLAIMED = 'claimed'  # Datum value to mark a case as claimed
+FUZZY_PROPERTIES = "fuzzy_properties"
```
I don't think this is used any more

proteusvacuum commented on corehq/apps/ota/views.py:
```diff
@@ -32,6 +42,70 @@ def restore(request, domain, app_id=None):
+            }
+        ).as_xml()
+        post_case_blocks([claim_case_block], {'domain': domain})
+    return HttpResponse(status=200)
```
I think this will respond 200 when other request types hit this endpoint too. Also, are there other errors states that could occur here that the mobile would want to know about?

orangejenny committed:
fix build

proteusvacuum commented on corehq/apps/ota/views.py:
```diff
@@ -32,6 +42,70 @@ def restore(request, domain, app_id=None):
+                'host': IndexAttrs(
+                    case_type=host_type,
+                    case_id=host_id,
+                    relationship='extension',
```
there is a CASE_INDEX_EXTENSION constant that I've been using elsewhere.

proteusvacuum commented on corehq/apps/ota/views.py:
```diff
@@ -32,6 +42,70 @@ def restore(request, domain, app_id=None):
+    # TODO: use request.session to ensure only one extension case if multiple claims for the same beneficiary
+    #       come from the same AWW.
+    if request.method == 'POST':
+        host_id = request.POST['case_id']
```
I could potentially see all of this going into a helper function `claim_case` that would live near where cases get closed, etc (I think in `cleanup.py` or something, although that name doesn't make sense any more if this would live there).

proteusvacuum commented on corehq/apps/ota/views.py:
```diff
@@ -32,6 +42,70 @@ def restore(request, domain, app_id=None):
+    search_es = CaseSearchES()
+    search_es = search_es.domain(domain)
+    search_es = search_es.case_type(case_type)
+    config = CaseSearchConfig(domain=domain).config
```
I think you need to do `CaseSearchConfig.objects.get(domain=domain)` here, otherwise you'll always end up with a new `CaseSearchConfig` object. You should also validate that the domain has case search enabled.

czue committed:
Merge pull request #11264 from dimagi/manage

Manage

czue committed:
Merge pull request #11242 from dimagi/clone_domain

management command to clone a domain

biyeun committed:
Merge pull request #11262 from lwyszomi/b3-care-pathways

B3 care pathways

kaapstorm committed:
project settings case search config ui

kaapstorm committed:
view added

biyeun committed:
catch full html pages being sent back

Commit:
Ability to export parent case ids

Commit:
add relationship and end paren

Commit:
exclude checking credit lines when subscription is suppressed

orangejenny committed:
fix sort_fields array

orangejenny committed:
unused import

orangejenny committed:
Update staging.yaml

Commit:
different style

Commit:
remove FutureWarning

Commit:
[ci skip] remove already implemented TODO

NoahCarnahan committed:
rewrite

snopoke committed:
working rebuild test cases

Commit:
add help text for the indices

snopoke committed:
uncomment todo asserts

snopoke committed:
pep8

benrudolph committed:
Merge pull request #11267 from dimagi/misc-cleanup

Misc cleanup

proteusvacuum committed:
Merge pull request #11269 from dimagi/better-index-repr

better CommCareIndex repr

Commit:
Merge pull request #11272 from dimagi/ignore-suppressed-subscriptions

exclude checking credit lines when subscription is suppressed

kaapstorm committed:
add get_or_none method to CaseSearchConfig model

kaapstorm committed:
view passed current values

kaapstorm committed:
initialize form values

kaapstorm committed:
post stub, drop bad imports

kaapstorm committed:
Manage properties better

NoahCarnahan committed:
Merge pull request #11271 from dimagi/export-caseids

Ability to export parent case ids in main table

kaapstorm committed:
view post

kaapstorm committed:
use Python naming convention where convenient

NoahCarnahan committed:
Merge pull request #11266 from dimagi/labels-or-ids-exports

useLabels and useIds for export display

sravfeyn committed:
Merge pull request #11275 from dimagi/remove-old-todo

 remove already implemented exports TODO

benrudolph committed:
Merge pull request #11276 from dimagi/rebuild

working rebuild test cases

benrudolph committed:
Merge pull request #11274 from dimagi/remove-warning

remove FutureWarning

benrudolph committed:
Merge pull request #11257 from dimagi/dbl-mgmt-form-names

Display double management form options in correct language

benrudolph committed:
Merge pull request #11265 from dimagi/scheduled-reports-js

Fix "schedule a report" 500

benrudolph committed:
Merge pull request #11249 from dimagi/purge

(trigger warning) Purge

benrudolph committed:
Merge pull request #11270 from dimagi/app-manager-html-response

catch full html pages being sent back on App manager error

czue committed:
Merge pull request #11231 from dimagi/sql-cloudcare

Cloudcare SQL

czue committed:
compare value to string

snopoke committed:
Merge pull request #11235 from dimagi/dogstatsd-gunicorn

add statsd host

snopoke committed:
Merge pull request #11278 from dimagi/mc-fix

compare value to string

kaapstorm committed:
jslint

kaapstorm committed:
Only post "case_type" and "properties" fuzzy property attributes

kaapstorm committed:
Misplaced comment

kaapstorm committed:
caseTypes doesn't need to be observable because it doesn't change

czue committed:
use font awesome chevrons

kaapstorm committed:
create CaseSearchConfigJSON instance correctly

czue committed:
add current month submissions to tables

czue committed:
return empty string instead of None when no change in chevron

kaapstorm committed:
No need for a wrapped view model

kaapstorm committed:
Handle zero-length fuzzy properties

kaapstorm commented:
Thanks for the feedback @proteusvacuum. I'll get on it as soon as module GUI is done.

NoahCarnahan committed:
lint

NoahCarnahan committed:
[ci skip] remove comment

kaapstorm committed:
Pass is_search_enabled to template

kaapstorm committed:
searchViewModel

kaapstorm committed:
module UI

czue committed:
add user information popovers to project health report

czue committed:
flip emphasis on delta and absolute numbers

czue committed:
[ci skip] lint

orangejenny committed:
submodules

proteusvacuum committed:
Use MultiTopicCheckpointEventHandler

The pillow was failing to run because it thought there was only one
topic

proteusvacuum committed:
Make section_name a parent prop

proteusvacuum committed:
prefactor: Don't add url params if there are none

Commit:
Merge pull request #11279 from dimagi/project-health

add current month information to project health report

esoergel committed:
Use unicode literals

Commit:
add NumericFilter

Commit:
add tests

Commit:
add NumericFilter to UI

biyeun committed:
management command to fix incorrect billables calculated prior to march 1, 2016 on the india server

NoahCarnahan committed:
Use hqDefine for chart.js

NoahCarnahan committed:
Add charts.clear function

NoahCarnahan committed:
Hide chart if greater than 25 categories

Commit:
update dimagi utils

Commit:
remove usages of expect_GET

NoahCarnahan committed:
Default page size to 25

esoergel committed:
Show css error class properly

esoergel committed:
Also error if the first row is left blank

[ci skip]

esoergel committed:
Always use "" to represent missing value [ci skip]

orangejenny committed:
Merge pull request #11282 from dimagi/unicode-problems

Use unicode literals

calellowitz committed:
Merge pull request #11283 from dimagi/numeric-range-filter

Numeric range filter for mobile UCR

calellowitz committed:
Merge pull request #11273 from dimagi/date-sort

Fix sorting for SMS billables report

NoahCarnahan committed:
Merge pull request #11244 from dimagi/revert-11243-revert-11234-migrate-export-management-cmd

Revert "Revert "Migrate export management cmd""

esoergel committed:
Merge pull request #11285 from dimagi/ucr-chart-limit

Report builder UCR chart category limit

proteusvacuum committed:
Hide save report button if it is admin report

Can't save preferences without a domain

proteusvacuum committed:
DeviceLogDomainFilter

gets all the domains

proteusvacuum committed:
Refactor DeviceLogDetailsReport to be subclassable

proteusvacuum committed:
Device log soft assert report

biyeun committed:
use more up-to-date version of clearing tab caches

esoergel committed:
Fix where user_data and location_id don't match

proteusvacuum committed:
Lint [ci skip]

orangejenny committed:
Merge pull request #11284 from dimagi/sms-fee-retro

FIX SMS billables on india before 3/1/2016

czue committed:
[ci skip] update help site link

http://manage.dimagi.com/default.asp?223689

czue committed:
Merge pull request #11281 from dimagi/multiple-change-feeds

Use MultiTopicCheckpointEventHandler

czue committed:
Merge pull request #11289 from dimagi/set-locations

Fix where user_data and location_id don't match

czue committed:
always use touchforms case list filters in cloudcare on sql

Commit:
remove unused variable

benrudolph committed:
Merge pull request #11290 from dimagi/help-site-fix

[ci skip] update help site link

benrudolph committed:
Merge pull request #11288 from dimagi/mobile-soft-assert-logs

Mobile soft assert logs

czue committed:
rename "project health" --> "project performance"

czue committed:
fix typo

snopoke committed:
Merge pull request #11291 from dimagi/cloudcare-sql-hack

always use touchforms case list filters in cloudcare on sql

czue committed:
unused import

czue committed:
rename tests

czue committed:
pull out user reindex test into its own class

czue committed:
split functionality out of test

czue committed:
make change providers take a DB instead of a doucment class

czue committed:
add v2 user reindexer and test

czue committed:
add test for webuser reindex

czue committed:
define USER_INDEX_INFO in mapping file

czue committed:
add doc types for commcare users and web users

czue committed:
alphabetize tests

czue committed:
add basic test for user pillow

czue committed:
don't require doc_prep_fn

czue committed:
don't fail hard initializing multi-topic feeds if some topics don't exist

czue committed:
add kafka-based user pillow and test

czue committed:
remove unused document_store argument from ConstructedPillow

czue committed:
create new topics

czue committed:
user new pillow for v2 reindexing

czue committed:
remove unnecessary pillow declarations from tests

czue committed:
use index info and utils instead of pillow

czue committed:
use new user reindexer

czue committed:
remove document_store from super call

czue committed:
[ci skip] imports

czue committed:
include docs in reindexers

czue committed:
update reindexing docs

czue committed:
rename ChangeProvider.iter_changes --> iter_all_changes

to avoid being confused with the pillow and change_feed methods

czue committed:
push useful functionality of PaginateViewLogHandler into default class

sravfeyn committed:
add require_GET

snopoke committed:
Merge pull request #11294 from dimagi/user-pillow

add (not hooked up yet) kafka pillow for ES user index

emord committed:
vellum:9b40dd61b61c25170c323c97bef216f1d8cd3fc1

czue committed:
Merge pull request #11296 from dimagi/csrf2.2

add require_GET

snopoke committed:
Merge pull request #11293 from dimagi/project-health

Project *performance* report wording tweaks

Commit:
changes in issue column

snopoke committed:
Merge pull request #11287 from dimagi/invalidate-app-cache

use more up-to-date version of clearing tab caches

czue committed:
delete old user reindexer and test

czue committed:
split out function

czue committed:
don't fail hard in delete_all_users if there are soft-deleted users

czue committed:
add test for soft deletion

snopoke committed:
Merge pull request #11297 from dimagi/user-pillow-cutover

use new user reindexer

emord committed:
update staging.yaml

[ci skip]

czue committed:
include document_store in couch change objects

czue committed:
use get_document

czue committed:
remove test for reindexer that isn't used, add test for pillow itself

czue committed:
delete reindexer that we aren't using anymore

we'll need to add this functionality back if we want to do it in the
future, but the current implementation seems worth killing since the
new version definitely will not rely on iterating through all forms
in the database one at a time

czue committed:
print

emord committed:
Merge pull request #11188 from dimagi/fix-xmlnss

Fix forms with "undefined" xmlns

snopoke committed:
Merge pull request #11300 from dimagi/user-pillow

remove legacy reindexer, add test

dimagimon commented on corehq/apps/app_manager/views/modules.py:
```diff
@@ -562,6 +565,7 @@ def edit_module_detail_screens(request, domain, app_id, module_id):
     persist_tile_on_forms = params.get("persistTileOnForms", None)
     pull_down_tile = params.get("enableTilePullDown", None)
     case_list_lookup = params.get("case_list_lookup", None)
+    search_properties = params.get("search_properties")
```
F841 local variable 'search_properties' is assigned to but never used

dimagimon commented on corehq/apps/app_manager/static/app_manager/js/detail-screen-config.js:
```diff
@@ -196,6 +196,60 @@ hqDefine('app_manager/js/detail-screen-config.js', function () {
+            var self = this;
+            self.name = ko.observable(name);
+            self.label = ko.observable(label);
+        }
```
Missing semicolon. (semi)

dimagimon commented on corehq/apps/app_manager/static/app_manager/js/detail-screen-config.js:
```diff
@@ -1068,6 +1126,7 @@ hqDefine('app_manager/js/detail-screen-config.js', function () {
                             containsFixtureConfiguration: (columnType == "short" && COMMCAREHQ.toggleEnabled('FIXTURE_CASE_SELECTION')),
                             containsFilterConfiguration: columnType == "short",
                             containsCaseListLookupConfiguration: (columnType == "short" && COMMCAREHQ.toggleEnabled('CASE_LIST_LOOKUP')),
+                            containsSearchConfiguration: columnType == "short",
```
Expected '===' and instead saw '=='. (eqeqeq)

dimagimon commented on corehq/apps/app_manager/views/modules.py:
```diff
@@ -167,7 +169,8 @@ def _get_basic_module_view_context(app, module):
         'valid_parent_modules': _get_valid_parent_modules(app, module),
         'child_module_enabled': toggles.BASIC_CHILD_MODULE.enabled(app.domain)
-                                 and module.doc_type != "ShadowModule"
+                                 and module.doc_type != "ShadowModule",
```
E131 continuation line unaligned for hanging indent

czue committed:
check for null docs

esoergel committed:
Merge pull request #11299 from lwyszomi/opm-next-changes

changes in issue column

Commit:
remove unused variable

Commit:
[cleanup] organize imports

Commit:
[cleanup] pep8

Commit:
fix community range end date

Commit:
update test to specify end date with no coverage

proteusvacuum commented on corehq/apps/ota/views.py:
```diff
@@ -32,6 +42,70 @@ def restore(request, domain, app_id=None):
+
+    search_es = CaseSearchES()
+    search_es = search_es.domain(domain)
+    search_es = search_es.case_type(case_type)
```
just thought of this, but we should currently limit this query to return a maximum number of results (maybe 10 for now?)
e.g: `search_es.size(MAX_CASE_SEARCH_RESULTS)`

kaapstorm replied:
A very good thought. Added in a051765.

czue committed:
Merge pull request #11303 from dimagi/remove-expect_GET

Remove expect_GET

NoahCarnahan committed:
Merge pull request #11286 from dimagi/report-builder-config

Misc minor report builder js/css fixes

esoergel committed:
Don't include docs on a reduced view

Views without a `reduce.js` default to `reduce=False`, but then if you
add a `reduce.js` later, those view calls change the default to
`reduce=True`.  This means you have to manually update any view calls that
aren't explicit about `reduce`.

esoergel committed:
Process fewer users

I'm getting a memory leak from this somehow, but I do think it's also
processing more stuff than it needs to.

Commit:
download jar from jenkins

benrudolph committed:
update staging.yaml

[ci skip]

benrudolph committed:
update staging.yaml

[ci skip]

benrudolph committed:
Merge pull request #11309 from dimagi/couch-annoyance

Don't include docs on a reduced view

Commit:
update staging.yaml

[ci skip]

benrudolph committed:
update staging.yaml

[ci skip]

Commit:
update staging.yaml

[ci skip]

benrudolph committed:
Merge pull request #11306 from dimagi/remove-test-var

remove unused variable

benrudolph committed:
Merge pull request #11301 from dimagi/unknown-users

remove unknown users reindexer

czue committed:
Merge pull request #11310 from dimagi/set-locations

Process fewer users on migration

czue committed:
Merge pull request #11307 from dimagi/fix-comm-sub-length

Fix community subscription length

czue committed:
Merge pull request #11305 from dimagi/remove-unused-variable

remove unused variable

emord committed:
Merge pull request #11311 from dimagi/fetch-formplayer-jar

Download formplayer jar from jenkins

kaapstorm committed:
No point in returning more than 10 results

kaapstorm committed:
Appeasing dimagimon

kaapstorm committed:
Save POSTed search properties

kaapstorm committed:
Initialize view model with current values

kaapstorm committed:
FB223512: Iterate properties, not search_config

dimagimon commented:
There are 6 errors:

* corehq/apps/hqadmin/management/commands/clone_domain.py, line 128 - W293 blank line contains whitespace
* corehq/apps/hqadmin/management/commands/clone_domain.py, line 50 - E125 continuation line with same indent as next logical line
E123 closing bracket does not match indentation of opening bracket's line
* corehq/apps/reports/views.py, line 996 - W293 blank line contains whitespace
* corehq/apps/hqadmin/management/commands/clone_domain.py, line 124 - W293 blank line contains whitespace
* corehq/apps/users/dbaccessors/all_commcare_users.py, line 49 - E301 expected 1 blank line, found 0
* corehq/apps/hqadmin/management/commands/clone_domain.py, line 134 - W293 blank line contains whitespace

kaapstorm commented:
Closing this PR against `uatbc/case-search-es`, and opening a new one against `master`.
